### PR TITLE
Migrate class-based assertion type components to functional

### PR DESCRIPTION
**Pull request by rnemes**
## Bug / Requirement Description
Official website (react.dev) states Component class as a legacy API and recommends defining components as functions instead of classes.
Functional components are simpler and less verbose than class components. They are also generally faster to render and easier to test.

## Solution description
Assertion and its underlaying components are functional (AssertionHeader. NotImplemented, BasicAssertion, CopyButton, etc...)
Snapshots should not be impacted as functionality won't change.

## Checklist:
- [ ] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [ ] News fragment present for release notes
- [x] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: testplan/web_ui/testing/src/AssertionPane/Assertion.js [hunk 1]
```diff
@@ -109,79 +79,77 @@ class Assertion extends Component {
...
+        isOpen={props.expand === EXPAND_STATUS.EXPAND}
+        className={css(styles.collapseDiv)}
+        style={{ paddingRight: isAssertionGroup ? null : "1.25rem" }}
+      >
+        <ErrorBoundary>
+          <CardBody
+            className={css(
+              isAssertionGroup
+                ? styles.groupCardBody
+                : styles.assertionCardBody
```
Comment (M6AI) [inline testplan/web_ui/testing/src/AssertionPane/Assertion.js:142]: While we are at refactoring a question (not keen on doing anything about it independent of the answer): are the best practices around CSS? Initially, I was used to external CSS, but here we use aphrodite (if I am not mistaken) to turn styles objects into CSS that is passed to the components.

File: testplan/web_ui/testing/src/AssertionPane/AssertionHeader.js [hunk 2]
```diff
@@ -13,190 +13,178 @@ library.add(faLayerGroup);
...
+    props.assertion.utc_time === undefined ? (
+      <span className={css(styles.cardHeaderAlignRight)}>
+        <FontAwesomeIcon // Should be a nested assertion group
+          size="sm"
+          key="faLayerGroup"
+          icon="layer-group"
+          className={css(styles.icon)}
+        />
+      </span>
+    ) : timeInfoArray.length === 0 ? (
```
Comment (M6AI) [inline testplan/web_ui/testing/src/AssertionPane/AssertionHeader.js:64]: Can this be truthy/falsy in JS too?

File: testplan/web_ui/testing/src/AssertionPane/AssertionTypes/BasicAssertion.js [hunk 3]
```diff
@@ -27,60 +27,59 @@ import { prepareBasicContent } from "./basicAssertionUtils";
...
-        </Row>
-        <Row>
-          <Col lg="12" className={css(styles.contentSpan)}>
-            {postContent}
-          </Col>
-        </Row>
-      </Fragment>
-    );
-  }
+  console.log("BasicAssertion has been rendered.");
```
Comment (M6AI) [inline testplan/web_ui/testing/src/AssertionPane/AssertionTypes/BasicAssertion.js:42]: Intentional or just local use and left in?

**Review by M6AI — COMMENTED** [review]
For now, looks good to me, some comments left.

File: testplan/web_ui/testing/src/AssertionPane/AssertionTypes/BasicAssertion.js (hunk 3 above)
Comment (rnemes) [reply testplan/web_ui/testing/src/AssertionPane/AssertionTypes/BasicAssertion.js:42]: Just for testing purposes, will be removed in final version.

File: testplan/web_ui/testing/src/AssertionPane/AssertionTypes/CopyButton.js [hunk 4]
```diff
@@ -1,21 +1,20 @@
...
-      <CopyToClipboard text={this.props.value} options={{ asHtml: true }}>
-        <Button outline color="secondary" size="sm" active={false}>
-          Copy
-        </Button>
-      </CopyToClipboard>
-    );
-  }
+function CopyButton({ value }) {
+
+  console.log("CopyButton has been rendered.");
```
[the remark this answers is not in the document]
Comment (rnemes) [reply testplan/web_ui/testing/src/AssertionPane/AssertionTypes/CopyButton.js:10]: Just for testing purposes, will be removed in final version.

File: testplan/web_ui/testing/src/AssertionPane/Assertion.js (hunk 1 above)
Comment (rnemes) [reply testplan/web_ui/testing/src/AssertionPane/Assertion.js:142]: We can solve that under a separate task.

File: testplan/web_ui/testing/src/AssertionPane/AssertionHeader.js (hunk 2 above)
Comment (rnemes) [reply testplan/web_ui/testing/src/AssertionPane/AssertionHeader.js:69]: I believe this component needs a proper refactoring under a separate task.

**Pull request merged**